libs/api/libbpf_program.cpp

```diff
@@ -25,6 +25,7 @@ bpf_load_program_xattr(const struct bpf_load_program_attr* load_attr, char* log_
         return libbpf_err(-EINVAL);
     }
 
+#if !defined(CONFIG_BPF_JIT_DISABLED) && !defined(CONFIG_BPF_INTERPRETER_DISABLED)
```

Comment:
nit: why do we want to block this at API layer also if it is already blocked in ebpfcore? Anyone can modify and recompile this dll to override this.

Reply:
Will having the checks in user mode block us to test the ebpfcore code where it fails the jit and interpret IOCTL calls for native_only case?

Reply:
answer: just reducing footprint.

Reply:
> nit: why do we want to block this at API layer also if it is already blocked in ebpfcore? Anyone can modify and recompile this dll to override this.

footprint reduction, and also changes the error return from "invalid value" to the more specific "not supported" meaning.

Reply:
> Will having the checks in user mode block us to test the ebpfcore code where it fails the jit and interpret IOCTL calls for native_only case?

The UM unit tests test that code path.